src/ec2/model.ts

```diff
@@ -4,29 +4,93 @@
+    }
+
+    protected createSsmSdkClient(): SsmClient {
+        return new SsmClient(this.regionCode)
     }
 
-    private async handleStartSessionError(err: AWS.AWSError): Promise<void> {
-        const failureMessage =
-            "SSM: Failed to start session with target instance. Common reasons include: \n 1. SSM Agent not installed on instance. \n 2. The required IAM instance profile isn't attached to the instance.  \n 3. Session manager setup is incomplete."
-        await vscode.window.showErrorMessage(failureMessage)
+    protected createEc2SdkClient(): Ec2Client {
+        return new Ec2Client(this.regionCode)
+    }
 
-        throw new ToolkitError('Start Session Failed. ')
+    protected createIamSdkClient(): DefaultIamClient {
+        return new DefaultIamClient(this.regionCode)
+    }
+
+    protected async getAttachedPolicies(instanceId: string): Promise<IAM.attachedPoliciesListType> {
+        try {
+            const IamRole = await this.ec2Client.getAttachedIamRole(instanceId)
+            const iamResponse = await this.iamClient.listAttachedRolePolicies(IamRole!.Arn!)
+            return iamResponse.AttachedPolicies!
+        } catch (err) {
+            return []
+        }
+    }
+
+    public async hasProperPolicies(instanceId: string): Promise<boolean> {
+        const attachedPolicies = (await this.getAttachedPolicies(instanceId)).map(policy => policy.PolicyName!)
+        const requiredPolicies = ['AmazonSSMManagedInstanceCore', 'AmazonSSMManagedEC2InstanceDefaultPolicy']
+
+        return requiredPolicies.every(policy => attachedPolicies.includes(policy))
+    }
+
+    public async handleStartSessionError(err: AWSError, selection: Ec2Selection): Promise<Error> {
+        const isInstanceRunning = await this.ec2Client.isInstanceRunning(selection.instanceId)
+        const generalErrorMessage = `Unable to connect to target instance ${selection.instanceId} on region ${selection.region}. `
+        const hasProperPolicies = await this.hasProperPolicies(selection.instanceId)
+
+        if (!isInstanceRunning) {
+            throw new ToolkitError(
+                generalErrorMessage +
+                    'Ensure the target instance is running and not currently starting, stopping, or stopped.',
+                { code: 'EC2SSMStatusError' }
+            )
+        }
+
+        if (!hasProperPolicies) {
+            throw new ToolkitError(
+                generalErrorMessage + 'Ensure the IAM role attached to the instance has the proper policies.',
+                {
+                    code: 'EC2SSMPermissionError',
+                    documentationUri: vscode.Uri.parse(
+                        'https://docs.aws.amazon.com/systems-manager/latest/userguide/session-manager-getting-started-instance-profile.html'
+                    ),
+                }
+            )
+        }
+
+        throw new ToolkitError(
+            'Ensure SSM is running on target instance. For more information see the documentation.',
```

Comment:
The docs button is enough. Keep messages very dense (but clear), to avoid overflow.
```suggestion
            'Is SSM running on the target instance?',
```